test/micro/org/openjdk/bench/java/net/URLEncodeDecode.java

```diff
@@ -51,78 +52,140 @@
 @Fork(value = 3)
 public class URLEncodeDecode {
 
-    @Param("1024")
-    public int count;
+    private static final int COUNT = 1024;
 
     @Param("1024")
     public int maxLength;
 
-    @Param("3")
-    public long mySeed;
+    /**
+     * Percentage of strings that will remain unchanged by an encoding/decoding (0-100)
+     */
+    @Param({"0", "75", "100"})
+    public int unchanged;
+
+    /**
+     * Percentage of chars in changed strings that cause encoding/decoding to happen (0-100)
+     */
+    @Param({"6"})
+    public int encodeChars;
 
     public String[] testStringsEncode;
     public String[] testStringsDecode;
     public String[] toStrings;
 
-    @Setup
+    @Setup()
     public void setupStrings() {
-        char[] tokens = new char[((int) 'Z' - (int) 'A' + 1) + ((int) 'z' - (int) 'a' + 1) + ((int) '9' - (int) '1' + 1) + 5];
+        char[] encodeTokens = new char[] { '[', '(', ' ', '\u00E4', '\u00E5', '\u00F6', ')', '='};
+        char[] tokens = new char[('Z' - 'A' + 1) + ('z' - 'a' + 1) + ('9' - '0' + 1) + 4];
         int n = 0;
-        tokens[n++] = '0';
-        for (int i = (int) '1'; i <= (int) '9'; i++) {
-            tokens[n++] = (char) i;
+        for (char c = '0'; c <= '9'; c++) {
+            tokens[n++] = c;
         }
-        for (int i = (int) 'A'; i <= (int) 'Z'; i++) {
-            tokens[n++] = (char) i;
+        for (char c = 'A'; c <= 'Z'; c++) {
+            tokens[n++] = c;
         }
-        for (int i = (int) 'a'; i <= (int) '<'; i++) {
-            tokens[n++] = (char) i;
+        for (char c = 'a'; c <= 'z'; c++) {
+            tokens[n++] = c;
         }
         tokens[n++] = '-';
         tokens[n++] = '_';
         tokens[n++] = '.';
-        tokens[n++] = '*';
+        tokens[n] = '*';
 
-        Random r = new Random(mySeed);
```

Comment:
SplittableRandom allow to have perfectly reproducible sequences

Reply:
Care to elaborate? `Random` with an explicit seed should be more than enough to have perfectly reproducible sequences for our case (`SplittableRandom` uses a different algorithm by default with a longer period, but I can't see how this'd be relevant to this case.)

```
 * If two instances of {@code Random} are created with the same
 * seed, and the same sequence of method calls is made for each, they
 * will generate and return identical sequences of numbers.
```

Reply:
Even with the same seed only with large numbers you get statistically a similar distribution,m (not the same values!), but splittable random is actually producing the same exact sequences. You care about confusing branch predictors here I suppose (or the JIT, maybe too) but not to add noise or variance with fewer samples nor you are interested to the security part of it (cause random use the chaos generated securely by the OS/HW iirc).

Reply:
I have to check in my notes why I used splittable rmd and will update this thread :)